### DISCUSSION TITLE
Failed to build TensorFlow’s C++ interface

### DISCUSSION THREAD

**Discussion by qdgp** [deepmodeling/deepmd-kit#426] [Q&A] [answered]
I'am tring to build TensorFlow’s C++ interface by this document (https://deepmd.readthedocs.io/en/latest/install-tf.2.3.html). My GCC version is 7.3.0, and my conda list is 

> _libgcc_mutex             0.1                        main  
> bazel                     3.1.0                he6710b0_0  
> blas                      1.0                         mkl  
> ca-certificates           2021.1.19            h06a4308_1  
> certifi                   2020.12.5        py37h06a4308_0  
> cudatoolkit               10.1.243             h6bb024c_0  
> cudatoolkit-dev           10.1.243             h516909a_3    conda-forge
> cudnn                     7.6.5                cuda10.1_0  
> intel-openmp              2020.2                      254  
> keras-preprocessing       1.1.2                    pypi_0    pypi
> ld_impl_linux-64          2.33.1               h53a641e_7  
> libffi                    3.3                  he6710b0_2  
> libgcc-ng                 9.1.0                hdf63c60_0  
> libstdcxx-ng              9.1.0                hdf63c60_0  
> mkl                       2020.2                      256  
> mkl-service               2.3.0            py37he8ac12f_0  
> mkl_fft                   1.3.0            py37h54f3939_0  
> mkl_random                1.1.1            py37h0573a6f_0  
> ncurses                   6.2                  he6710b0_1  
> numpy                     1.19.2           py37h54aff64_0  
> numpy-base                1.19.2           py37hfa32c7d_0  
> openjdk                   8.0.152              h7b6447c_3  
> openssl                   1.1.1j               h27cfd23_0  
> pip                       21.0.1           py37h06a4308_0  
> python                    3.7.10               hdb3f193_0  
> python_abi                3.7                     1_cp37m    conda-forge
> readline                  8.1                  h27cfd23_0  
> setuptools                52.0.0           py37h06a4308_0  
> six                       1.15.0           py37h06a4308_0  
> sqlite                    3.35.2               hdfb4753_0  
> tk                        8.6.10               hbc83047_0  
> wheel                     0.36.2             pyhd3eb1b0_0  
> xz                        5.2.5                h7b6447c_0  
> zlib                      1.2.11               h7b6447c_3

After execute ''bazel build -c opt --verbose_failures //tensorflow:libtensorflow_cc.so
", the error is 

> ERROR: /mnt/gs18/scratch/users/gepei/tensorflow/tensorflow/cc/BUILD:545:1: Linking of rule '//tensorflow/cc:ops/lookup_ops_gen_cc' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command 
>   (cd /mnt/ufs18/home-183/gepei/.cache/bazel/_bazel_gepei/515900c424239e830302083d92d41b3a/execroot/org_tensorflow && \
>   exec env - \
>     LD_LIBRARY_PATH=/opt/software/binutils/2.30-GCCcore-7.3.0/lib:/opt/software/GCCcore/7.3.0/lib/gcc/x86_64-pc-linux-gnu/7.3.0:/opt/software/GCCcore/7.3.0/lib64:/opt/software/GCCcore/7.3.0/lib \
>     PATH=/mnt/home/gepei/anaconda3/envs/tensorflow_GPU/bin:/opt/software/binutils/2.30-GCCcore-7.3.0/bin:/opt/software/GCCcore/7.3.0/bin:/mnt/home/gepei/anaconda3/condabin:/usr/lib64/qt-3.3/bin:/opt/software/core/lua/lua/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/local/hpcc/bin:/usr/lpp/mmfs/bin:/opt/ibutils/bin:/opt/puppetlabs/bin:/opt/dell/srvadmin/bin \
>     PWD=/proc/self/cwd \
>   external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc @bazel-out/k8-opt-exec-50AE0418/bin/tensorflow/cc/ops/lookup_ops_gen_cc-2.params)
> Execution platform: @local_execution_config_platform//:platform
> bazel-out/k8-opt-exec-50AE0418/bin/_solib_local/_U_S_Stensorflow_Scc_Cops_Slookup_Uops_Ugen_Ucc___Utensorflow/libtensorflow_framework.so.2: undefined reference to `std::allocator<long long>::allocator()'
> collect2: error: ld returned 1 exit status
> Target //tensorflow:libtensorflow_cc.so failed to build
> INFO: Elapsed time: 1151.735s, Critical Path: 227.31s
> INFO: 5812 processes: 5812 local.
> FAILED: Build did NOT complete successfully

I have tried lots of times by different environment but the error is same. The tensorflow source code version is 2.3.0. Did I forget to install some packages? Thanks a lot.

**Comment by njzjz** [comment]
I suggest you to search on https://github.com/tensorflow/tensorflow
You may open a new issue there.

**Comment by qdgp** [comment, marked as answer]
Fixed by changing gcc from 7.3.0 to 7.5.0
